src/start_server.ts

```diff
@@ -181,6 +161,179 @@ function startWithPort(userOptions: ServerOptions) {
       });
     }
   }
+}
+
+/**
+ * Determines whether a protocol requires HTTPS
+ * @param {string} protocol Protocol to evalulate.
+ * @returns {boolean}
+ */
+function isHttps(protocol: string) {
+  return ['https/1.1', 'https', 'http2', 'h2'].indexOf(protocol) > -1;
+}
+
+/**
+ * Gets the URLs for the main and component pages
+ * @param {ServerOptions} options
+ * @returns {{serverUrl: {protocol: (string|string), hostname: string, port: string}, componentUrl: url.Url}}
+ */
+function getServerUrls(options: ServerOptions) {
+  let serverUrl = {
+    protocol: isHttps(options.protocol) ? 'https' : 'http',
+    hostname: options.hostname,
+    port: `${options.port}`,
+  };
+  let componentUrl: url.Url = Object.assign({}, serverUrl);
+  componentUrl.pathname = `components/${options.packageName}/`;
+  return {serverUrl, componentUrl};
+}
+
+/**
+ * Handles server-ready tasks (prints URLs and opens browser)
+ * @param {ServerOptions} options
+ */
+function handleServerReady(options: ServerOptions) {
+  let {serverUrl, componentUrl} = getServerUrls(options);
+  console.log(`Files in this directory are available under the following URLs
+    applications: ${url.format(serverUrl)}
+    reusable components: ${url.format(componentUrl)}`);
+  openBrowser(options, serverUrl, componentUrl);
+}
+
+/**
+ * Generates a TLS certificate for HTTPS
+ * @returns {Promise<{}>} Promise of {serviceKey: string, certificate: string}
+ */
+function createTLSCertificate() {
+  return new Promise<{}>((resolve, reject) => {
+    console.log('Generating TLS certificate...');
+    pem.createCertificate({days: 1, selfSigned: true}, function(err: any, keys: any) {
+      if (err) {
+        reject(err);
+      } else {
+        writeFile('.cert.pem', keys.certificate);
```

Comment:
You need to wait for these Promises to complete: `Promise.all(writeFile(...), writeFile(...)).then((_) => resolve(keys));`